packages/core/src/connectivity/scp-cf/csrf-token-header.ts

```diff
@@ -24,22 +24,24 @@ const logger = createLogger({
 export async function buildCsrfHeaders<T extends HttpRequestConfig>(
   destination: Destination | DestinationNameAndJwt,
   requestConfig: Partial<T>
-): Promise<Record<string, string>> {
+): Promise<Record<string, any>> {
   const csrfHeaders = await makeCsrfRequest(destination, requestConfig);
   validateCsrfTokenResponse(csrfHeaders);
-  return filterNullishValues({
-    ...getHeader('x-csrf-token', csrfHeaders),
-    cookie: buildCookieHeaderValue(getHeaderValue('set-cookie', csrfHeaders))
+  return pickNonNullishIgnoreCase({
+    ...pickIgnoreCase(csrfHeaders, 'x-csrf-token'),
+    cookie: buildCookieHeaderValue(
+      pickValueIgnoreCase(csrfHeaders, 'set-cookie')
+    )
   });
 }
 
 function makeCsrfRequest<T extends HttpRequestConfig>(
   destination: Destination | DestinationNameAndJwt,
   requestConfig: Partial<T>
 ): Promise<Record<string, any>> {
-  const fetchHeader = !getHeaderValue(
-    'x-csrf-token',
-    requestConfig.headers
+  const fetchHeader = !pickValueIgnoreCase(
+    requestConfig.headers,
+    'x-csrf-token'
```

Comment:
should we perhaps have a const class with all the keys, or perhaps there is already a lib for that?

Reply:
Not that I am aware of. Might have a nice touch to it (maybe not necessarily a class ;) ), but from the top of my head I don't see a big benefit. If you still think this should be done maybe create a ticket and let's discuss in the refinement?